src/io.rs

```diff
@@ -20,8 +20,11 @@ impl<W: Write> WriteExt for W {
     }
 
     fn write_char(&mut self, c: char) -> io::Result<usize> {
-        let utf8 = c.encode_utf8();
-        self.write(utf8.as_slice())
+        let mut utf8 = [0; 4];
+        match c.encode_utf8(&mut utf8) {
+            Some(len) => self.write(&utf8[0..len]),
+            None => panic!("<char>.encode_utf8 error!"),    // A buffer of length four is large enough to encode any char
```

Comment:
It did return io error, because `panic!` will never be executed(the comment explained)